### PR TITLE
Ratelimit file downloads

### PR DESCRIPTION
**Pull request by ying-stripe**

### CODE REVIEW

File: sequins.go [hunk 1]
```diff
@@ -100,6 +107,13 @@ func (s *sequins) init() error {
 	}
 	s.workQueue = workqueue.NewWorkQueue(maxLoads)
 
+	// Set up token bucket for download bandwidth throttling
+	maxDownloadBandwidth := s.config.MaxDownloadBandwidthMBPerSecond * 1024 * 1024
```
Comment (wei-stripe) [inline sequins.go]: Can we just not introduce this artificial limit and just pass nil to bucket?

In the downloader, if the bucket is nil, we can skip the reader wrap operation.

File: vendor/vendor.json [hunk 2]
```diff
@@ -1659,6 +1665,12 @@
 			"path": "golang.org/x/net/lex/httplex",
 			"revision": "dd2d9a67c97da0afa00d5726e28086007a0acce5",
 			"revisionTime": "2017-02-23T02:43:42Z"
+		},
+		{
+			"checksumSHA1": "538I8ghUdvZa25NxgMUDpq14Qic=",
+			"path": "gopkg.in/check.v1",
```
Comment (vasi-stripe) [inline vendor/vendor.json]: Do we need to include this? It looks like juju/ratelimit only uses this for its tests, which we shouldn't be running, right?

**Comment by vasi-stripe** [comment]
LGTM

**Comment by vasi-stripe** [comment]
(modulo nitpicks)

File: sequins.go (hunk 1 above)
Comment (ying-stripe) [reply sequins.go]: Okay. 

In my opinion, we should throttling by default. The downloading part has lower priority. It should not compete with the serving part on bandwidth and CPU.

File: vendor/vendor.json (hunk 2 above)
Comment (ying-stripe) [reply vendor/vendor.json]: I used govendor to add juju/ratelimit. It added this automatically. If I remove it, next time when somebody runs govendor again, may it be added back? I will check it.

File: build.go [hunk 3]
```diff
@@ -187,7 +198,7 @@ func (vs *version) sparkeyDownload(src, dst, fileType string, transform func(io.
 		}
 	}()
 
-	var trStream io.Reader = stream
+	var trStream = rateLimitedStream
```
Comment (wei-stripe) [inline build.go]: can be simplified to:
`trStream =: vs.rateLimitedDownloadStream(stream)`

**Comment by wei-stripe** [comment]
lgtm

**Pull request merged**